lib/internal/console/constructor.js

```diff
@@ -432,7 +432,12 @@ const consoleMethods = {
 
   assert(expression, ...args) {
     if (!expression) {
-      args[0] = `Assertion failed${args.length === 0 ? '' : `: ${args[0]}`}`;
+      // https://console.spec.whatwg.org/#assert step 4.2:
+      if (typeof args[0] === "string") {
+        args[0] = `Assertion failed: ${args[0]}`
+      } else {
+        ArrayPrototypeUnshift(args, "Assertion failed")
+      }
```

Comment:
```suggestion
      let baseMessage = 'Assertion failed';
      if (args.length > 0) {
        baseMessage += ':';
      }
      ArrayPrototypeUnshift(args, "Assertion failed");
```

That way it's always consistent and I would not add the documentation reference as it might change it's location without the comment being updated.